### ISSUE TITLE
Inductor should respect rand op order in eager function

### ISSUE THREAD

**Issue by BoyuanFeng** [pytorch/pytorch#181136]
The following code fails with compile. In eager code, rng order is `a -> b -> c`. In compiled code, rng order is `b -> c -> a`.

```python
import torch

def fn():
    a = torch.randn(2, 4, 4, 4, device="cuda")
    b = torch.randn(2, 3, 4, 4, device="cuda")
    c = torch.randn(4, 3, 3, 3, device="cuda")
    y = torch.nn.functional.conv2d(b, c, padding=1)
    z = a + y
    return z

f_c = torch.compile(fn)

torch.manual_seed(42)
eager_out = fn()

torch.manual_seed(42)
eager_out1 = fn()

torch.manual_seed(42)
compile_out = f_c()

torch.testing.assert_close(eager_out, eager_out1)   # PASS
torch.testing.assert_close(compile_out, eager_out)    # FAIL
```

cc @pbelevich @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

**Comment by liqiangxl** [comment]
Two root causes here:

**1. Scheduler reordering (`fallback_random=True`)** — Inductor's scheduler reorders `randn` fallback calls based on data dependencies, consuming global RNG state out of program order. Fix in #181245: chain `nondeterministic_seeded` nodes with `WeakDep` to preserve order.

**2. Different RNG mechanism (`fallback_random=False`, default)** — Inductor fuses multiple `randn` calls into one Triton kernel. Since Triton kernels can't access the global CUDA RNG, each op gets its own Philox stream via `inductor_seeds`. This produces different values from eager by design. `align_random_eager` ([#178843](https://github.com/pytorch/pytorch/pull/178843)) partially addresses this for `rand`/dropout but not `randn`, and `rand_eager_kernel` has two issues that prevent bare `torch.rand` from matching eager:
- **Complement**: returns `1.0 - x` instead of `x` (needed for dropout since `native_dropout` CUDA uses `curand < (1-p)` while the decomposition uses `rand > p` — can be fixed by changing the decomposition to `rand < (1-p)` and removing the complement)
- **Element ordering**: maps elements to Philox outputs in stride-4 order (element 0→philox[0], element 4→philox[1]) instead of contiguously

**Path forward:**
1. Merge #181245 — fixes `fallback_random=True` scheduler ordering
2. Fix `rand_eager_kernel` complement + element ordering so `align_random_eager` works for bare `torch.rand`, not just dropout
3. Extend `align_random_eager` to support `randn` (requires reproducing CUDA's Box-Muller transform in Triton)
